service/controller/app/v1/resource/kubeconfigfinalizer/create.go

```diff
@@ -0,0 +1,67 @@
+package kubeconfigfinalizer
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/giantswarm/microerror"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/giantswarm/app-operator/service/controller/app/v1/key"
+)
+
+func (r *Resource) EnsureCreated(ctx context.Context, obj interface{}) error {
+	cr, err := key.ToCustomResource(obj)
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	if key.InCluster(cr) {
+		r.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("In-cluster used"))
+		r.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("no need to put the finalizer"))
+		return nil
+	}
+
+	name := key.KubecConfigSecretName(cr)
+	namespace := key.KubecConfigSecretNamespace(cr)
+
+	r.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("finding kubeconfig secret for app %#q", name))
```

Comment:
This is the app or the secret name? If this is the secret printing namespace would be useful. Printing app name here isn't very informative as this is already injected into `object` key in logs.

```suggestion
	r.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("finding kubeconfig secret for app %#q", name))
```

Reply:
This is `app` name actually so I would change it to `secret` name with namespace. Thanks.